### PR TITLE
Agency Settings & Metric Config: Action Required treatment

### LINKED ISSUE #672: [Frontend] Agency Settings & Metric Config: Action Required treatment

### PR DESCRIPTION
**Pull request by mxosman**
## Description of the change

Adds red text color treatment to Metric Configuration "Action Required" title and adds red "Action Required" text to agency settings with no values entered.

https://github.com/Recidiviz/justice-counts/assets/59492998/2c85f288-c467-4989-8dcf-ec3e01169776

## Related issues

Closes #672 

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: publisher/src/components/Settings/AgencySettingsDescription.tsx [hunk 1]
```diff
@@ -55,6 +55,11 @@ const AgencySettingsDescription: React.FC<{
     currentAgencySettings?.find(
       (setting) => setting.setting_type === "PURPOSE_AND_FUNCTIONS"
     )?.value || "";
+  const isAgencySettingConfigured = Boolean(
```
Comment (lilidworkin) [inline publisher/src/components/Settings/AgencySettingsDescription.tsx:58]: nit but isn't this a little redundant with the line above, which does almost the same thing? could we say `isAgencySettingConfigured = purposeAndFunctionsSetting !== ""`

File: publisher/src/components/Settings/AgencySettingsURL.tsx [hunk 2]
```diff
@@ -55,6 +55,11 @@ const AgencySettingsUrl: React.FC<{
     currentAgencySettings?.find(
       (setting) => setting.setting_type === "HOMEPAGE_URL"
     )?.value || "";
+  const isAgencySettingConfigured = Boolean(
```
Comment (lilidworkin) [inline publisher/src/components/Settings/AgencySettingsURL.tsx:58]: same ? as above

**Review by lilidworkin — APPROVED** [review]
Looks great! Small nit inline

File: publisher/src/components/Settings/AgencySettingsDescription.tsx (hunk 1 above)
Comment (mxosman) [reply publisher/src/components/Settings/AgencySettingsDescription.tsx:58]: Ah - great call! Wow - talk about tunnel-vision... Will update!

**Pull request merged**